builder/oracle/oci/driver_oci.go

```diff
@@ -19,6 +23,30 @@ type driverOCI struct {
 	context       context.Context
 }
 
+var retryPolicy = &common.RetryPolicy{
+	MaximumNumberAttempts: 10,
+	ShouldRetryOperation: func(res common.OCIOperationResponse) bool {
+		var e common.ServiceError
+		if errors.As(res.Error, &e) {
+			if e.GetHTTPStatusCode() == 429 || e.GetHTTPStatusCode() == 500 || e.GetHTTPStatusCode() == 503 {
+				return true
+			}
+		}
+		return false
+	},
+	NextDuration: func(res common.OCIOperationResponse) time.Duration {
+		x := uint64(res.AttemptNumber)
+		d := time.Duration(math.Pow(2, float64(atomic.LoadUint64(&x)))) * time.Second
+		j := time.Duration(rand.Float64()*(2000)) * time.Millisecond
+		w := d + j
+		return w
```

Comment:
it could be nice to implement the Exponential backoff function here: https://github.com/hashicorp/packer/blob/37652b05ebb08f990212b3b6751237e8138f093c/vendor/github.com/hashicorp/packer-plugin-sdk/retry/retry.go#L116-L119


Or to use that one: https://github.com/cenkalti/backoff/blob/v4/exponential.go

Reply:
Hm. I'm a bit back and forth about this. The python oci sdk has some default strategies and I think it includes an "exponential plus jitter" one (but haven't checked again now). And I think I remember it also has one that can easily be set for the whole client (i.e. all api calls). The go sdk though seems to have gotten a little less love when it comes to this. What I find on their end (only here and there, no standard strategy that's recommended in general) is this:

https://github.com/oracle/oci-go-sdk/blob/d1b4be36a7910cdb25ad13b801eeb507c41d8be9/common/retry_test.go#L37

```
		return time.Duration(math.Pow(float64(2), float64(response.AttemptNumber-1))) * time.Second
```

(which is pretty much what I suggest here, just without the jitter and starting a bit less steeply)
So this would be the more "native" approach, use the retry strategies the sdk has within.

The retry.go from the packer-plugin-sdk kind of does the same, but that would mean using the plain oci go sdk access (with getting all errors). It also seems mostly used for the polling of resources (i.e. the backoff between successful requests). That would be a somewhat larger change, I think. Changing all the retry and waiting code in the oci driver to use the functions from there.

I'm not saying one approach is better than the other. Just not sure which route to go. My first idea was to go with the sdk provided retry stuff (which I think aso the aws driver uses). Practically even the minimal (use switch, but no jitter) would solve the acute problem we face which is 429 stop some builds (if the user that is used for the build does a lot of api requests).

Reply:
Oh, I see. I think I like your approach better then !